r-vpn.tf

```diff
@@ -37,6 +37,27 @@ resource "azurerm_virtual_network_gateway" "public_virtual_network_gateway" {
     }
   }
 
+  dynamic "custom_route" {
+    for_each = var.additional_routes_to_advertise
+
+    content {
+      address_prefixes = var.additional_routes_to_advertise
+    }
+  }
+
+  dynamic "vpn_client_configuration" {
+    for_each = var.vpn_aad_client_configuration != null ? [var.vpn_aad_client_configuration] : []
+    iterator = vpn
+
+    content {
+      aad_audience         = vpn.value.aad_audience
+      aad_issuer           = vpn.value.aad_issuer
+      aad_tenant           = vpn.value.aad_tenant
+      address_space        = [vpn.value.address_space]
+      vpn_auth_types       = ["AAD"]
+      vpn_client_protocols = ["OpenVPN"]
```

Comment:
With hardcoded value, we can't use any other protocol.
If `AAD` is not compatible with anything else than `OpenVPN` you can use variable validation or `pre_condition` checks.

https://www.terraform.io/language/expressions/custom-conditions